code/game/objects/items/airbubble.dm

```diff
@@ -208,72 +222,88 @@
 	playsound(loc, "sound/items/[pick("rip1","rip2")].ogg", 100, 1)
 	break_open()
 	animate_shake()
+	desc += " <span class='danger'>It has hole in it! Maybe you shouldn't use it!</span>"
 	qdel(bar)
 
+// We are out finally, the bubble is ripped. So dump everything out from it. Especially air and user.
 /obj/structure/closet/airbubble/break_open()
 	ripped = TRUE
 	update_icon()
 	dump_contents()
+	var/datum/gas_mixture/t_air = get_turf_air()
+	t_air.merge(inside_air)
 
 // Change valve on internal tank
 /obj/structure/closet/airbubble/verb/set_internals(mob/user as mob)
 	set src in oview(1)
 	set category = "Object"
 	set name = "Set Internals"
 	if(!isnull(internal_tank))
-		visible_message("<span class='warning'>[user] is setting [src] internals.</span>")
-		user << "<span class='notice'>You are settting [src] internals.</span>"
+		user.visible_message(
+		"<span class='warning'>[user] is setting [src] internals.</span>",
+		"<span class='notice'>You are settting [src] internals.</span>"
+		)
 		if (!do_after(user, 2 SECONDS, act_target = src))
 			return
-		visible_message("<span class='warning'>[user] have set [src] internals.</span>")
-		user << "<span class='notice'>You set [src] internals.</span>"
+		user.visible_message(
+		"<span class='warning'>[user] have set [src] internals.</span>" ,
```

Comment:
shouldnt it be has set ?

Reply:
Well, I was assuming we use pronoun `they`?